qemu/tests/bridge_mirror.py

```diff
@@ -77,28 +76,34 @@ def check_tcpdump(output, src_ip, des_ip, ping_count):
                 reply_num += 1
 
         if request_num != ping_count or reply_num != ping_count:
-            logging.debug(
-                "Unexpect request or reply number. "
-                "current request number is: %d, "
-                "current reply number is: %d, "
-                "expected request and reply number is: %d. "
-                % (request_num, reply_num, ping_count))
+            logging.debug("Unexpect request or reply number. "
+                          "current request number is: %d, "
+                          "current reply number is: %d, "
+                          "expected request and reply number is: %d. "
+                          % (request_num, reply_num, ping_count))
             return False
         return True
 
+    netdst = params.get("netdst", "switch")
+    br_backend = utils_net.find_bridge_manager(netdst)
+    if not isinstance(br_backend, utils_net.Bridge):
+        test.cancel("Host does not use Linux Bridge")
+
     brname = params.get("private_bridge", "tmpbr")
     net_mask = params.get("net_mask", "24")
     login_timeout = int(params.get("login_timeout", "600"))
     stop_NM_cmd = params.get("stop_NM_cmd")
     stop_firewall_cmd = params.get("stop_firewall_cmd")
-    add_private_bridge_cmd = params.get("add_private_bridge_cmd")
     tcpdump_cmd = params.get("tcpdump_cmd")
     tcpdump_log = params.get("tcpdump_log")
     get_tcpdump_log_cmd = params.get("get_tcpdump_log_cmd")
     ping_count = int(params.get("ping_count"))
 
     error_context.context("Create a private bridge", logging.info)
-    process.system_output(add_private_bridge_cmd, shell=True)
+    br_backend.add_bridge(brname)
+    br_iface = utils_net.Interface(brname)
+    br_iface.up()
+    br_iface.promisc_on()
```

Comment:
Do you want to add this function to avocado-vt?

Reply:
If you mean add a function to "create br -> set it up -> set promisc on", I think we can use existing func as above.
If you mean "set promisc", it already exists in vt

Reply:
Yes, I mean add functions to create a private bridge in vt. Anyway it's up to you. Both are good to me.

Reply:
Since we have a func `utils_net.add_bridge()`, I thinks it's enough for our test to create a bridge we want. Thanks your advice.